src/components/sbb-dialog/sbb-dialog.tsx

```diff
@@ -274,6 +277,21 @@ export class SbbDialog implements AccessibilityProperties {
     }
   };
 
+  // Close the dialog on click of any element that has the 'sbb-dialog-close' attribute.
+  private _closeOnSbbDialogCloseClick(event: Event): void {
+    const target = event.target as HTMLElement;
+
+    // Check if the target is a submit element within a form
+    const closestForm =
+      target.getAttribute('type') === 'submit'
+        ? (hostContext('form', target) as HTMLFormElement)
+        : undefined;
```

Comment:
The closest form is only used inside the next if condition, I'd suggest to wrap it into it.